base/interactiveutil.jl

```diff
@@ -278,7 +278,8 @@ typesof(args...) = Tuple{map(a->(isa(a,Type) ? Type{a} : typeof(a)), args)...}
 gen_call_with_extracted_types(fcn, ex0::Symbol) = Expr(:call, fcn, Meta.quot(ex0))
 function gen_call_with_extracted_types(fcn, ex0)
     if isa(ex0, Expr) &&
-        any(a->(Meta.isexpr(a, :kw) || Meta.isexpr(a, :parameters)), ex0.args)
+          (any(a->(Meta.isexpr(a, :kw) || Meta.isexpr(a, :parameters)), ex0.args) ||
+           ex0.head == :call)
```

Comment:
I think this kills the branch in 296 completely. The main requirement for fixing https://github.com/JuliaLang/julia/issues/17469 is that the expression goes through the `expand` step in line 290. However, it's hard to predict the consequences of changes to this function.